Fix crash in small screens and refactor test (MacOS only)

SamuMazzi opened the pull request.

Kalmat commented:
Thank you!

If I am not wrong, the position and size values are still the same than in previous version, right? Is the problem with small screens solved and, just curious, how?

SamuMazzi commented:
Exactly, I left the same values, I just refactored a bit. Regarding the problem I just moved the window before making it bigger, so that it could have enough space to get larger.

Also I wanted to say that this piece of code `lowered = npw.lowerWindow()` doesn't return the correct value (in the test it actually doesn't get lowered)
Maybe this isn't the right place to tell it, but I didn't want to forget that

Kalmat commented:
By the way, the `test_move()` function is gorgeous! I didn't know that was even possible! You will never go to bed... without learning something new! HAHAHAHA!

I have adopted that solution for the rest of the OSs. Now all basic_* tests look the same.

Thank you!
